site/src/middleware.page.ts

```diff
@@ -5,18 +5,42 @@ import { NextResponse } from "next/server";
 
 import { SESSION_COOKIE_NAME } from "./lib/api/middleware/constants";
 
+const productionFrontendHost = process.env.NEXT_PUBLIC_FRONTEND_URL
+  ? new URL(process.env.NEXT_PUBLIC_FRONTEND_URL).host
+  : undefined;
+const productionSandboxHost = process.env.NEXT_PUBLIC_BLOCK_SANDBOX_URL
+  ? new URL(process.env.NEXT_PUBLIC_BLOCK_SANDBOX_URL).host
+  : undefined;
+
 export async function middleware(request: NextRequest) {
-  const { cookies } = request;
+  const url = new URL(request.url);
+
+  if (productionFrontendHost && productionSandboxHost) {
+    const openingBlockSandboxPage = Boolean(
+      url.pathname.match(/^\/@[\w_-]+\/blocks\/[\w_-]+\/sandboxed-demo$/),
+    );
 
-  // destroying session at `logout.api.ts` removes this cookie
-  const isLoggedIn = !!cookies.get(SESSION_COOKIE_NAME);
+    if (url.host === productionFrontendHost && openingBlockSandboxPage) {
+      const newUrl = new URL(url);
+      newUrl.host = productionSandboxHost;
+      return NextResponse.redirect(newUrl, 308);
+    }
```

Comment:
I feel like defining a function like this
```ts
const changeHostAndRedirect = (host: string) => {
  const newUrl = new URL(url);
  newUrl.host = host;
  return NextResponse.redirect(newUrl, 308);
};
```

and reusing it in both if's looks a bit easier-to-read

```ts
if (url.host === productionFrontendHost && openingBlockSandboxPage) {
  return changeHostAndRedirect(productionSandboxHost);
}

if (url.host === productionSandboxHost && !openingBlockSandboxPage) {
  return changeHostAndRedirect(productionFrontendHost);
}
```